File: tests/xcalablemp/gmove/C/correct/gmove_cp_1a1t_c.c
#include <stdio.h>
#include <math.h>
#include <xmp.h>

int n=8;
double a[n],b[n];
#pragma xmp nodes p(2)
#pragma xmp template tx(0:n-1)
#pragma xmp distribute tx(cyclic) onto p
#pragma xmp align a[i] with tx(i)
#pragma xmp align b[i] with tx(i)

int main(){

  int i,j;
  int myrank;
  double err;

  myrank=xmp_all_node_num();

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    a[i]=i+1;
  }

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    b[i]=0;
  }

#pragma xmp gmove
  b[3:4]=a[0:4];

  err=0.0;
#pragma xmp loop (i) on tx(i)
  for(i=3;i<7;i++){
      err=err+fabs(b[i]-(i-2));
  }

  printf("max error=%f,myrank=%d\n",err,myrank);

  return 0;

}

File: tests/xcalablemp/global-view/gmove/Fortran/correct/gmove_cp_1a1t_c.f90
program tgmove
integer i,n=8
integer a(n),b(n)
!$xmp nodes p(2)
!$xmp template tx(n)
!$xmp distribute tx(cyclic) onto p
!$xmp align a(i) with tx(i)
!$xmp align b(i) with tx(i)

!$xmp loop (i) on tx(i)
do i=1,n
  a(i)=i
end do

!$xmp loop (i) on tx(i)
do i=1,n
  b(i)=0
end do

!$xmp gmove
b(2:5)=a(5:8)

ierr=0
!$xmp loop (i) on tx(i)
do i=2,5
  ierr=ierr+abs(b(i)-(i+3))
end do

print *, 'max error=',ierr

contains
end
